### PR TITLE
Parity for local beanstalk snippet testing

### PR DESCRIPTION
**Pull request by spectorar**

### CODE REVIEW

**Comment by EmilyMcCarthy** [comment]
Looks like the build is broken, can we get that fixed?

File: lib/remote_payload.coffee [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  if reqQuery.f
+    execParams = {}
+    if reqQuery.params
+      try
+        execParams = JSON.parse(reqQuery.params)
+      catch e
+        execParams = reqQuery.params #as handled in platform
+  else
+    execParams = _.assign {}, (if _.isObject(reqPayload) then reqPayload else {}), reqQuery
+    delete execParams.apikey if execParams.apikey
```
Comment (EmilyMcCarthy) [inline lib/remote_payload.coffee:36]: don't we currently pass the apikey into the snippet as params?

File: lib/remote_payload.coffee [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  execParams
+
+create = (req) ->
+  reqPayload = getReqPayload req.payload, req.headers['content-type']
```
Comment (EmilyMcCarthy) [inline lib/remote_payload.coffee:45]: are all the headers lower cased automatically?

File: lib/remote_payload.coffee [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    response:
+      body:
+        request:
+          method: req.method.toUpperCase()
+          'content-type': req.headers['content-type']
+    session:
+      api_key: req.headers['x-cloudmine-apikey'] or req.query.apikey
+      app_id: req.params.appid
+      session_token: req.headers['x-cloudmine-sessiontoken'] or null
+      user_id: '[User ID not populated in local deployments]' #snippetConfig.userId -- needs platform request
```
Comment (EmilyMcCarthy) [inline lib/remote_payload.coffee]: This is remote request right? So not local, should have the user id

**Comment by EmilyMcCarthy** [comment]
This needs tests

File: lib/remote_payload.coffee (hunk 2 above)
Comment (spectorar) [inline lib/remote_payload.coffee:45]: yes

File: lib/remote_payload.coffee (hunk 3 above)
Comment (spectorar) [inline lib/remote_payload.coffee]: Local refers to the fact that you are testing locally. Do you think there is better wording? Local test deployments maybe?

File: lib/remote_payload.coffee (hunk 1 above)
Comment (spectorar) [inline lib/remote_payload.coffee:36]: Yes, this just removes it from the "params" object if it was provided in the query string. See below, api_key gets populated under the "session" object

**Comment by Splat** [comment]
Looks like there is some failing tests in Jenkins.

**Pull request merged**